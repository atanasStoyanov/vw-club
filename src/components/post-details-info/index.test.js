import React from 'react';
import renderer from 'react-test-renderer';
import PostDetailsInfo from '.';
import TestingEnvironmet from '../../test-utils/router';

jest.mock('../post-values', () => 'PostValues');

describe('PostDetailsInfo Component', () => {
    it('should render post details info with carModel="Test car" and description="Test desc"', () => {
        const post = {
            carModel: 'Test car',
            description: 'Test desc'
        }

        const tree = renderer.create(
            <TestingEnvironmet value={{
                loggedIn: true,
                user: {
                    id: '123'
                }
            }}>
                <PostDetailsInfo post={post} />
            </TestingEnvironmet>).toJSON();

        expect(tree).toMatchSnapshot();
    })
})