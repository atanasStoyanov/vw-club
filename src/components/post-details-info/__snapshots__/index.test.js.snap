// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailsInfo Component should render post details info with carModel="Test car" and description="Test desc" 1`] = `
<div>
  <div>
    <img
      alt="car"
      className="image"
      src="passat-icon.svg"
    />
  </div>
  <div
    className="info"
  >
    <h4
      className="model"
    >
      <small>
        Car model:
      </small>
       
      Test car
    </h4>
    <h3>
      Post Description
    </h3>
    <p>
      Test desc
    </p>
    <PostValues />
  </div>
</div>
`;
